Cargo.toml

```diff
@@ -70,6 +73,12 @@ notify = { version = "5.0.0-pre.13", optional = true }
 console-subscriber = { version = "0.1", optional = true, features = ["parking_lot"] }
 const_format = "0.2"
 atty = "0.2"
+quinn = { version = "0.8.0", optional = true}
+rustls = { version = "0.20", default-features = false, features = ["quic"], optional = true }
+rustls-pemfile = { version = "0.2.1", optional = true }
+openssl = { version = "0.10.38", optional = true }
```

Comment:
I'm absolutely not a fan of pkcs. I just wanted to stick with the existing pattern.

Reply:
Looking into `native-tls` I notice that they also use `openssl` for parsing p12.  So I don't see the benefit of using a different crate when the method I'm using is already being compiled in via `native-tls`

Reply:
it matters when you compile with feature `quic` and without `tls` :)
`openssl` is really a pain to compile. Dropping it will possibly make the feature combination `server,client,quic` available on more platforms than `server,client,tls` is.
`openssl` already causes trouble for cross compiling `rathole`  and I have to degrade `cross` to work around. Someone requests for a official risc-v release but I can't do it simply because of the cross compiling stuff about `openssl`.
`tls` chooses to use `openssl` only because it supports IP as hostname and that's many people want.  I wish to shift all tls dependencies to `rustls` at certain point.

Reply:
So, I just tried out the only really promising crate that does not end up relying on `openssl`: https://github.com/hjiayz/p12

Unfortunately it choked on the p12 file I created myself. As I said, I am absolutely not a fan of PKCS12 - precisely because you never known quite which libraries, are compatible with with PKCS12 formats. Unless of course you choose a huge lib like `openssl` that has _almost_ everything covered. (See https://github.com/rustls/rustls/issues/150 where they decline to add support n rusttls)
I don't really want to pull in a half baked pkcs12 parser that only works for some users.

I would suggest the following in a separate PR:

 - implement reading certs and keys from PEM files directly, as an alternative to supplying pfx file. 
 - create a new "pfx" feature that pulls in `openssl` for users who want it and enables support for pfx in QUIC transport.

Reply:
Fair. We can leave it in a separate PR

Reply:
Surprisingly, `openssl` seems to make CI fail. This doesn't happen with `tls` feature, which also implicitly includes `openssl`. Any ideas?

Reply:
Yeah, i think `native-tls` depends on `openssl` when compiled on Linux platform. When compiled on Mac or Windows, I think it uses platform specific deps instead.

The way I have setup `cargo.toml` forces `openssl` to be compiled regardless of platform.

I would need to figure out how to do this per platform scoping to fix the builds.

I may have to, at least temporarily, restrict usage of pfx in quic to Linux only.

Reply:
BTW, I'm open to the change of specifying the certificate in other ways, if pkcs12 is troublesome to work with.

Reply:
Pkcs is often very troublesome. Usually the problem stems from jkcs keystores that pretend to be p12 stores, yet not entirely compatible.
Also a problem the other way around, when a Java app expects a jkcs but is provided with an openssl p12.

I will implement reading from pem files. 

Can we scope quic-pfx to be available on Linux only?

Reply:
> I will implement reading from pem files.
> Can we scope quic-pfx to be available on Linux only?

If we have an alternative way to use the certificate, I think we can drop pfx entirely in this PR, as you previously suggested

Reply:
`native-tls` unfortunately will not support PEM files only. 
Only certs can be loaded from PEM files, the keys must still come from pfx.